packages/js/src/generators/library/library.ts

```diff
@@ -177,6 +177,18 @@ function updateTsConfig(tree: Tree, options: NormalizedSchema) {
   });
 }
 
+function addBabelConfig(tree: Tree, options: NormalizedSchema) {
+  const filename = '.babelrc';
+  const babelrc = {
+    presets: [['@nrwl/web/babel', { useBuiltIns: 'usage' }]],
```

Comment:
question/issue: if I don't have `@nrwl/web` installed and this is the default (`.babelrc`), can my library still build/test properly?

Reply:
> question/issue: if I don't have `@nrwl/web` installed and this is the default (`.babelrc`), can my library still build/test properly?

I setup a local npm registry and run the following tests:

## 1. Express Application

`package.json`

```json
"devDependencies": {
    "@nrwl/cli": "20.0.0",
    "@nrwl/eslint-plugin-nx": "20.0.0",
    "@nrwl/express": "^20.0.0",
    "@nrwl/jest": "20.0.0",
    "@nrwl/js": "^20.0.0",
    "@nrwl/linter": "20.0.0",
    "@nrwl/node": "20.0.0",
    "@nrwl/workspace": "20.0.0",
    "@types/express": "4.17.13",
    "@types/jest": "27.4.1",
    "@types/node": "16.11.7",
    "@typescript-eslint/eslint-plugin": "~5.18.0",
    "@typescript-eslint/parser": "~5.18.0",
    "eslint": "~8.12.0",
    "eslint-config-prettier": "8.1.0",
    "jest": "27.5.1",
    "nx": "20.0.0",
    "prettier": "^2.5.1",
    "ts-jest": "27.1.4",
    "ts-node": "9.1.1",
    "typescript": "~4.6.2"
  },
  "dependencies": {
    "express": "4.17.2",
    "tslib": "^2.3.0"
  }
```

Created a `js` library and imported within an express application with typescript

### Result

No compilation error

## 1. Node Application

`package.json`

```json
  "devDependencies": {
    "@nrwl/cli": "20.0.0",
    "@nrwl/eslint-plugin-nx": "20.0.0",
    "@nrwl/jest": "20.0.0",
    "@nrwl/js": "^20.0.0",
    "@nrwl/linter": "20.0.0",
    "@nrwl/node": "^20.0.0",
    "@nrwl/workspace": "20.0.0",
    "@types/jest": "27.4.1",
    "@types/node": "16.11.7",
    "@typescript-eslint/eslint-plugin": "~5.18.0",
    "@typescript-eslint/parser": "~5.18.0",
    "eslint": "~8.12.0",
    "eslint-config-prettier": "8.1.0",
    "jest": "27.5.1",
    "nx": "20.0.0",
    "prettier": "^2.5.1",
    "ts-jest": "27.1.4",
    "ts-node": "9.1.1",
    "typescript": "~4.6.2"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  }
```

<img width="2273" alt="Screenshot 2022-05-02 at 16 44 23" src="https://user-images.githubusercontent.com/45793869/166254124-78e8af1b-4f4c-4e7d-9941-c8cd70b5f9dd.png">

Created a `js` library and imported it within an nodejs application with typescript

### Result

No compilation error

Reply:
@nartc is my first time digging into the NX source code so please tell me if I am mistaken, but from an initial look it seems that the `@nrwl/web` packages is installed as part of `@nrwl/workspaces`.

Are there scenarios where you could bootstrap NX monorepos without the `@nrwl/workspaces` ?

Anyway, I don't want to make dangerous guess, so I am happy to refactor to have the `.babelrc` as an opt-in flag if necessary.

Let me know your thoughts

Reply:
@floroz `@nrwl/web` is not a dependency of `@nrwl/workspace` 
<img width="570" alt="image" src="https://user-images.githubusercontent.com/25516557/166259151-7fcbca28-4b23-41f7-804e-0d21dcd002d4.png">

You can have an Nx workspace without `nrwl/web`. If your setup does not have `nrwl/web` and it compiles properly (both build and test), then it should be good. Can you confirm that you don't have `nrwl/web` in your setup shown above?

Reply:
<img width="1289" alt="Screenshot 2022-05-02 at 17 40 58" src="https://user-images.githubusercontent.com/45793869/166263459-bb44390b-d0f7-4c53-8558-7004d140a69e.png">

Reply:
Yeah, it is a little awkward for people who's building CLI only applications/libraries. They don't need `nrwl/web`. A `.babelrc` (a config file) that contains `nrwl/web` feels off and confusing. I think we can make it a bit better

- Rename `skipBabelRc` to `includeBabelRc` and remove its default value (aka it will be `undefined`).
- In the generator, we check `package.json` whether `nrwl/web` exists. If it does, then we set `includeBabelRc` to `true` ONLY IF it's not already set (aka if the users set this flag explicitly, we respect their choice)
- Then we check `includeBabelRc == true` to generate the `.babelrc` along with the library.

Thoughts?

Reply:
Thanks for the feeedback @nartc.

I like your approach, feels flexible enough and doesn't get in the way of the developer.

I'll start refactoring the code within this PR 👍